src/graphql/queries/EntryValues.ts

```diff
@@ -1,22 +1,165 @@
 import { gql } from 'react-apollo';
 import { decode } from '../../services/UrlStringCreator';
+import { ComponentConfig } from '../../typings/schema';
+import { DataSets, checkTypes, Query } from '../../typings/schema';
 
-const QUERY_ENTRY_VALUES = ({ match, currentCollection, values }) => {
-    values = values || [];
+// `type: never` makes the type checker report an error if the case switch does not handle all types
+function checkUnknownComponent(type: never) {
+    console.error(`Type ${(type as ComponentConfig).type} is not handled!`);
+}
+
+function getTitleProp(typeId: string, otherCollections: Array<{collectionId: string, summaryProperties: {title?: {value: string}}}>): string {
+    for (let i = 0; i < otherCollections.length; i++) {
+        if (otherCollections[i].collectionId === typeId) {
+            const title = otherCollections[i].summaryProperties.title;
+            if (title) {
+                return '.' + title.value;
+            } else {
+                return '.uri';
+            }
+        }
+    }
+    return '.uri';
+}
+
+export function makeDefaultViewConfig(properties: Array<{ isList: boolean, isValueType: boolean, name: string, referencedCollections: { items: string[] } }>, summaryProperties: { title?: { value: string } }, otherCollections: Array<{ collectionId: string, summaryProperties: { title?: { value: string } } }>): Array<ComponentConfig> {
```

Comment:
Create an interface for 'OtherCollections'. Double declaration of otherCollections. The getTitleProp also uses it.

```
interface OtherProperty {
    collectionId: string;
    summaryProperties: SummaryProperties // from schema.ts
}
```